spec/basic_examples_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "hypothesis"
+
+def expect_failure(&block)
+  expect(&block).to raise_exception(RSpec::Expectations::ExpectationNotMetError)
+end
+
+RSpec.configure do |c|
+  c.include(Hypothesis)
+  c.include(Hypothesis::Providers)
+end
+
+RSpec.describe "some basic hypothesis tests" do
+  it "should think integer addition is commutative" do
+    hypothesis do 
```

Comment:
While this works, I think it's more idiomatic rspec to pass a `:hypothesis` message to the `it` block (`it "foo", :hypothesis do`), and have that wrap the body in a hypothesis block. I don't know what the corresponding would be in Minitest, so having explicit hypothesis blocks might be a safe bet.

This raises a question: should it be possible to pass hypothesis to a context? Should the following be possible?

```ruby
context "adding", :hypothesis do
  let(:x) { given integers }
  let(:y) { given integers }

  it "is has 0 as the identity" do
    expect(x + 0).to eq(x)
    expect(0 + x).to eq(x)
  end
end
```

Reply:
this is something users can configure themselves using a simple RSpec setup.

Reply:
I hadn't been planning on supporting it in contexts - I imagine it would interact interestingly with rspec because it would end up running the same it block many times.

In terms of :hypothesis vs hypothesis, I think it would be nice to try to keep the core interface as test runner agnostic as possible - as you say, minitest is also a thing. Right now this has no dependency on rspec at all. I expect I'll need to relax that a bit for reporting purposes, but keeping the code the same between different testing libraries is definitely preferable.